meson.build

```diff
@@ -253,6 +253,12 @@ configure_file(
   configuration : conf
 )
 
+default_sysconfdir = get_option('sysconfdir')
+if default_sysconfdir == 'etc'
+  message('sysconfdir of etc makes no sense, using /etc')
+  default_sysconfdir = '/etc'
+endif
```

Comment:
Is this for getting around a particular issue you were seeing?
It makes me think it might be worth trying to workaround meson's innane default of using how it handles the defaults when it comes to prefixes too (https://github.com/mesonbuild/meson/issues/1637)

Like perhaps detect if something is set to /usr/local for prefix and override sysconfdir and localstatedir for those situations if they're not explicitly set on the meson command.

Reply:
It moves the fallback to the build system, making the code simpler.